src/main/scala/amba/apb/ToTL.scala

```diff
@@ -0,0 +1,91 @@
+// See LICENSE.SiFive for license details.
+
+package freechips.rocketchip.amba.apb
+
+import Chisel._
+import freechips.rocketchip.config.Parameters
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.tilelink._
+import freechips.rocketchip.util.MaskGen
+
+case class APBToTLNode()(implicit valName: ValName) extends MixedAdapterNode(APBImp, TLImp)(
+  dFn = { case APBMasterPortParameters(masters) =>
+    TLClientPortParameters(clients = masters.map { m =>
+      TLClientParameters(name = m.name, nodePath = m.nodePath)
+    })
+  },
+  uFn = { mp => APBSlavePortParameters(
+    slaves = mp.managers.map { m =>
+      APBSlaveParameters(
+        address       = m.address,
+        resources     = m.resources,
+        regionType    = m.regionType,
+        executable    = m.executable,
+        nodePath      = m.nodePath,
+        //TODO why and how should I do this adjust for APB?
+        // Also, shouldn't I use supportsPutPartial because of PSTRB?
+        supportsWrite = m.supportsPutPartial.intersect(TransferSizes(1, mp.beatBytes)),
+        supportsRead  = m.supportsGet.intersect(TransferSizes(1, mp.beatBytes)))},
+    beatBytes = mp.beatBytes)
+  })
+
+class APBToTL()(implicit p: Parameters) extends LazyModule
+{
+  val node = APBToTLNode()
+
+  
+  lazy val module = new LazyModuleImp(this) {
+    (node.in zip node.out) foreach { case ((in, edgeIn), (out, edgeOut)) =>
+      val beatBytes = edgeOut.manager.beatBytes
+
+      //-----------------------------------------------
+      // Flow Control
+      // We don't want to handle the case where the data is returned immediately
+      // on the same cycle that we send the transaction. Require and verify.
+      require (edgeOut.manager.minLatency >= 1)
+      assert (!(in.psel && !in.penable && out.d.valid))
+
+      val in_flight_reg = RegInit(false.B)
```

Comment:
One possible issue I noticed:  The APB spec has an active-low reset, but this implementation appears to be active-high.